packages/build-utils/src/types.ts

```diff
@@ -49,15 +49,31 @@ export interface Config {
 export type HasField = Array<
   | {
       type: 'host';
-      value: string;
+      value: MatchableValue;
     }
   | {
       type: 'header' | 'cookie' | 'query';
       key: string;
-      value?: string;
+      value?: MatchableValue;
     }
 >;
 
+type MatchableValue =
```

Comment:
This is just copy-pasta'd from the type change in https://github.com/vercel/vercel/pull/13375

As the type error confirmed, these types must in fact match.